generic_config_updater/gu_common.py

```diff
@@ -136,6 +139,22 @@ def validate_config_db_config(self, config_db_as_json):
 
         return True, None
 
+    def validate_field_operation(self, old_config, target_config):
+        """
+        Some fields in ConfigDB are restricted and may not allow third-party addition, replacement, or removal. 
+        Because YANG only validates state and not transitions, this method helps to JsonPatch operations/transitions for the specified fields. 
+        """
+        patch = jsonpatch.JsonPatch.from_diff(old_config, target_config)
+        
+        # illegal_operations_to_fields_map['remove'] yields a list of fields for which `remove` is an illegal operation 
+        illegal_operations_to_fields_map = {'add':[],
+                                            'replace': [],
+                                            'remove': ['/PFC_WD/GLOBAL/POLL_INTERVAL', '/PFC_WD/GLOBAL']}
+        for operation, field_list in illegal_operations_to_fields_map.items():
+            for field in field_list:
+                if any(op['op'] == operation and field in op['path'] for op in patch):
```

Comment:
Hi @isabelmsft , I think we should use an exact match for field comparison: `field == op['path']`
In current pfcwd yang model, the GLOBAL table only contians one field: POLL_INTERVAL. But I notice it could also contains other fields. https://github.com/sonic-net/sonic-utilities/blob/master/tests/mock_tables/asic0/config_db.json#L119

If we use field **in** op['path'], it will block all other fields change(such as BIG_RED_SWITCH).

Reply:
thank you for the suggestion, changed to exact match